twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php

```diff
@@ -380,45 +380,53 @@ public function authenticate($username, $password) {
         $errorCode = 0;
         $res = 0;
 
-        try {
-            $client = $this->httpClientService->newClient();
-            $res = $client->get($url, $options);
-            $body = $res->getBody();
-            $body = json_decode($body);
+        if ($url) {
```

Comment:
My PHP is very rusty -- If I I understand correctly: in case the password is empty, ``$url`` is undefined. In this case, does ``if ($url) {} else {} `` just evaluate the ``else`` block, or does it raise an error? I'm asking because it would raise an error in Python, but not in JS (I think), but I'm not sure about PHP :)

Reply:
This will not throw. `$url` will be coerced into `false`, similar to how it would work in JS.

Reply:
Thanks for the clarification!